File: app/src/main/java/com/zaze/demo/component/readpackage/adapter/ReadPackageAdapter.java
package com.zaze.demo.component.readpackage.adapter;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.zaze.aarrepo.commons.base.ZBaseApplication;
import com.zaze.aarrepo.commons.base.adapter.BaseItemHolder;
import com.zaze.aarrepo.commons.base.adapter.BaseUltimateAdapter;
import com.zaze.aarrepo.utils.AppUtil;
import com.zaze.aarrepo.utils.StringUtil;
import com.zaze.demo.R;
import com.zaze.demo.model.entity.PackageEntity;

import java.util.Collection;
import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Description :
 *
 * @author : ZAZE
 * @version : 2017-04-17 - 17:21
 */
public class ReadPackageAdapter extends BaseUltimateAdapter<PackageEntity, ReadPackageAdapter.PackageHolder> {

    public ReadPackageAdapter(Context context, Collection<PackageEntity> data) {
        super(context, data);
    }

    @Override
    public PackageHolder getViewHolder(View view, boolean isItem) {
        return new PackageHolder(view, isItem);
    }

    @Override
    public int getViewLayoutId() {
        return R.layout.list_item_package;
    }

    @Override
    public void onBindViewHolder(PackageHolder holder, PackageEntity value, int position) {
        String packageName = StringUtil.parseString(value.getPackageName());
        holder.itemPackageTv.setText(StringUtil.format("%s : %s", AppUtil.getAppName(ZBaseApplication.getInstance(), packageName, ""), packageName));
        Drawable drawable = AppUtil.getAppIcon(ZBaseApplication.getInstance(), packageName);
        if (drawable == null) {
            drawable = getDrawable(R.drawable.ic_launcher);
        }
        holder.itemPackageIv.setImageDrawable(drawable);

    }

    @Override
    public long generateHeaderId(int position) {
        return 0;
    }

    @Override
    public RecyclerView.ViewHolder onCreateHeaderViewHolder(ViewGroup parent) {
        return null;
    }

    @Override
    public void onBindHeaderViewHolder(RecyclerView.ViewHolder holder, int position) {
    }


    public String getPackageListStr() {
        List<PackageEntity> packageEntityList = getDataList();
        StringBuilder builder = new StringBuilder();
        for (PackageEntity entity : packageEntityList) {
            if (builder.length() > 0) {
                builder.append(",");
            }
            builder.append(entity.getPackageName());
        }
        return builder.toString();
    }

    class PackageHolder extends BaseItemHolder {
        @Bind(R.id.item_package_iv)
        ImageView itemPackageIv;
        @Bind(R.id.item_package_tv)
        TextView itemPackageTv;


        PackageHolder(View itemView, boolean isItem) {
            super(itemView, isItem);
            ButterKnife.bind(this, itemView);
        }

        @Override
        protected void initView(View itemView) {

        }
    }
}

File: app/src/main/java/com/zaze/demo/component/readpackage/adapter/ReadPackageAdapter.kt
package com.zaze.demo.component.readpackage.adapter

import android.content.Context
import android.content.pm.PackageManager
import android.graphics.drawable.Drawable
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.zaze.common.adapter.BaseRecyclerAdapter
import com.zaze.common.base.BaseApplication
import com.zaze.demo.R
import com.zaze.demo.debug.AppShortcut
import com.zaze.demo.debug.InvariantDeviceProfile
import com.zaze.utils.AppUtil
import com.zaze.utils.BmpUtil
import com.zaze.utils.ZStringUtil

/**
 * Description :

 * @author : ZAZE
 * *
 * @version : 2017-04-17 - 17:21
 */
class ReadPackageAdapter(context: Context, data: Collection<AppShortcut>) : BaseRecyclerAdapter<AppShortcut, ReadPackageAdapter.PackageHolder>(context, data) {

    private val iconDpi: Int

    val iconSize = 72

    init {
        iconDpi = InvariantDeviceProfile().getLauncherIconDensity(iconSize)
    }

    override fun createViewHolder(convertView: View): PackageHolder {
        return PackageHolder(convertView)
    }

    override fun getViewLayoutId(): Int {
        return R.layout.app_recycle_item
    }

    override fun onBindView(holder: PackageHolder, value: AppShortcut, position: Int) {
        val packageName = ZStringUtil.parseString(value.packageName)
        // --------------------------------------------------
        holder.itemAppNameTv.text = "应用名 : ${value.name}"
        holder.itemAppVersionCodeTv.text = "版本号 ：${value.versionCode}"
        holder.itemAppVersionNameTv.text = "版本名 ：${value.versionName}"
        holder.itemAppPackageTv.text = "包名 : $packageName"
        holder.itemAppDirTv.text = "路径 : ${value.sourceDir}"
        holder.itemAppSignTv.text = "签名 : ${value.signingInfo}"
        // --------------------------------------------------
        var drawable: Drawable? = null
        val application = AppUtil.getApplicationInfo(context, packageName)
        if (application != null) {
            val resources = try {
                context.packageManager.getResourcesForApplication(application)
            } catch (e: PackageManager.NameNotFoundException) {
                null
            }
            if (resources != null) {
                drawable = AppUtil.getAppIcon(resources, application.icon, iconDpi)
            }
        }
        if (drawable == null) {
            drawable = AppUtil.getAppIcon(BaseApplication.getInstance(), packageName)
        }
        if (drawable == null) {
            drawable = getDrawable(R.mipmap.ic_launcher)
        }
        holder.itemAppIv.setImageBitmap(BmpUtil.drawable2Bitmap(drawable, iconSize))
    }

    inner class PackageHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var itemAppIv: ImageView = itemView.findViewById(R.id.item_app_iv) as ImageView
        var itemAppNameTv: TextView = itemView.findViewById(R.id.item_app_name_tv) as TextView
        var itemAppPackageTv: TextView = itemView.findViewById(R.id.item_app_package_tv) as TextView
        var itemAppVersionCodeTv: TextView = itemView.findViewById(R.id.item_app_version_code_tv) as TextView
        var itemAppVersionNameTv: TextView = itemView.findViewById(R.id.item_app_version_name_tv) as TextView
        var itemAppDirTv: TextView = itemView.findViewById(R.id.item_app_dir_tv) as TextView
        var itemAppSignTv: TextView = itemView.findViewById(R.id.item_app_sign_tv) as TextView

    }
}
